pyomo/contrib/mindtpy/single_tree.py

```diff
@@ -657,256 +686,162 @@ def handle_lazy_subproblem_other_termination(
-                    'Fixed subproblem will be generated based on the incumbent solution of the main problem.'
-                )
-        elif main_mip_results.solver.termination_condition is tc.maxTimeLimit:
-            config.logger.info(
-                'Regularization problem failed to converge within the time limit.'
-            )
-            solve_data.results.solver.termination_condition = tc.maxTimeLimit
-        elif main_mip_results.solver.termination_condition is tc.unbounded:
-            config.logger.info(
-                'Regularization problem ubounded.'
-                'Sometimes solving MIQP using cplex, unbounded means infeasible.'
-            )
-        elif main_mip_results.solver.termination_condition is tc.unknown:
-            config.logger.info(
-                'Termination condition of the regularization problem is unknown.'
-            )
-            if main_mip_results.problem.lower_bound != float('-inf'):
-                config.logger.info('Solution limit has been reached.')
-                handle_main_optimal(main_mip, solve_data, config, update_bound=False)
-        else:
-            raise ValueError(
-                'MindtPy unable to handle regularization problem termination condition '
-                'of %s. Solver message: %s'
-                % (
-                    main_mip_results.solver.termination_condition,
-                    main_mip_results.solver.message,
-                )
-            )
-
     def __call__(self):
-        """This is an inherent function in LazyConstraintCallback in cplex.
+        """This is an inherent function in LazyConstraintCallback in CPLEX.
 
         This function is called whenever an integer solution is found during the branch and bound process.
         """
-        solve_data = self.solve_data
+        mindtpy_object = self.mindtpy_object
         config = self.config
         opt = self.opt
         main_mip = self.main_mip
+        mindtpy_object = self.mindtpy_object
 
-        if solve_data.should_terminate:
-            self.abort()
-            return
+        config.logger.debug(
+            "Solution source: %s (111 node_solution, 117 heuristic_solution, 119 mipstart_solution)".format(
+                self.get_solution_source()
+            )
+        )
 
-        self.handle_lazy_main_feasible_solution(main_mip, solve_data, config, opt)
+        # The solution found in MIP start process might be revisited in branch and bound.
+        # Lazy constraints separated when processing a MIP start will be discarded after that MIP start has been processed.
+        # This means that the callback may have to separate the same constraint again for the next MIP start or for a solution that is found later in the solution process.
+        # https://www.ibm.com/docs/en/icos/12.8.0.0?topic=SSSA5P_12.8.0/ilog.odms.cplex.help/refpythoncplex/html/cplex.callbacks.LazyConstraintCallback-class.htm
+        if (
+            self.get_solution_source() != 119
```

Comment:
Perhaps you could create an Enum-like data structure for these codes since you have to use them so often. I'm sure they won't change, but for the sake of readability, it might be nice to use descriptive names and just map them to the integers once.